File: storeJav/storeJav/src/test/java/com/vetleDemo/store/HomeControllerWebTest.java
package com.vetleDemo.store;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.test.web.servlet.MockMvc;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;

@WebMvcTest(HomeController.class)
class HomeControllerWebTest {

    @Autowired
    private MockMvc mockMvc;

    @Test
    void rootReturnsIndexHtmlViewName() throws Exception {
        mockMvc.perform(get("/"))
                .andExpect(status().isOk())
                .andExpect(view().name("index.html"));
    }
}

File: storeKot/src/test/kotlin/com/vetled/store/HomeControllerWebTest.kt
package com.vetled.store

import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.get

@WebMvcTest(HomeController::class)
class HomeControllerWebTest {

    @Autowired
    lateinit var mockMvc: MockMvc

    @Test
    fun `root returns index html view`() {
        mockMvc.get("/")
            .andExpect {
                status { isOk() }
                view { name("index.html") }
            }
    }
}

